test/Benchmarks/src/Table/Sorting.enso

```diff
@@ -0,0 +1,40 @@
+from Standard.Base import all
+from Standard.Base.Data.Index_Sub_Range import Sample
+import Standard.Base.Data.Time.Duration
+
+import Standard.Table
+from Standard.Table import Sort_Column_Selector
+from Standard.Table.Data.Aggregate_Column import all
+
+import Standard.Test.Bench
+
+type My
+    Data x
+
+    compare_to self other =
+        other.x . compare_to self.x
+
+vector_size = 100000
+iter_size = 20
+num_iterations = 10
+
+# The Benchmarks ==============================================================
+main =
```

Comment:
I've tried running the benchmarks locally and I'm getting the following results:
```
Table.order_by ints/iteration:0: 1370.20ms
Table.order_by ints/iteration:1: 1350.27ms
Table.order_by ints/iteration:2: 1302.77ms
Table.order_by ints/iteration:3: 1294.60ms
Table.order_by ints/iteration:4: 1328.23ms
Table.order_by ints/iteration:5: 1325.52ms
Table.order_by ints/iteration:6: 1342.88ms
Table.order_by ints/iteration:7: 1300.48ms
Table.order_by ints/iteration:8: 1306.90ms
Table.order_by ints/iteration:9: 1270.68ms
Vector.sort ints/iteration:0: 0.03ms
Vector.sort ints/iteration:1: 0.00ms
Vector.sort ints/iteration:2: 0.00ms
Vector.sort ints/iteration:3: 0.00ms
Vector.sort ints/iteration:4: 0.00ms
Vector.sort ints/iteration:5: 0.00ms
Vector.sort ints/iteration:6: 0.00ms
Vector.sort ints/iteration:7: 0.00ms
Vector.sort ints/iteration:8: 0.00ms
Vector.sort ints/iteration:9: 0.00ms
Table.order_by dates/iteration:0: 3603.56ms
Table.order_by dates/iteration:1: 3591.66ms
Table.order_by dates/iteration:2: 3616.44ms
Table.order_by dates/iteration:3: 3606.24ms
Table.order_by dates/iteration:4: 3552.48ms
Table.order_by dates/iteration:5: 3596.27ms
Table.order_by dates/iteration:6: 3630.64ms
Table.order_by dates/iteration:7: 3561.34ms
Table.order_by dates/iteration:8: 3585.17ms
Table.order_by dates/iteration:9: 3569.49ms
Vector.sort dates/iteration:0: 0.01ms
Vector.sort dates/iteration:1: 0.00ms
Vector.sort dates/iteration:2: 0.02ms
Vector.sort dates/iteration:3: 0.00ms
Vector.sort dates/iteration:4: 0.00ms
Vector.sort dates/iteration:5: 0.00ms
Vector.sort dates/iteration:6: 0.00ms
Vector.sort dates/iteration:7: 0.00ms
Vector.sort dates/iteration:8: 0.00ms
Vector.sort dates/iteration:9: 0.00ms
Table.order_by objects/iteration:0: 25684.12ms
Table.order_by objects/iteration:1: 25303.84ms
Table.order_by objects/iteration:2: 25170.91ms
Table.order_by objects/iteration:3: 25042.65ms
Table.order_by objects/iteration:4: 25231.87ms
Table.order_by objects/iteration:5: 25325.89ms
Table.order_by objects/iteration:6: 25196.74ms
Table.order_by objects/iteration:7: 25218.00ms
Table.order_by objects/iteration:8: 25338.44ms
Table.order_by objects/iteration:9: 25401.82ms
Vector.sort objects/iteration:0: 0.02ms
Vector.sort objects/iteration:1: 0.00ms
Vector.sort objects/iteration:2: 0.00ms
Vector.sort objects/iteration:3: 0.00ms
Vector.sort objects/iteration:4: 0.00ms
Vector.sort objects/iteration:5: 0.00ms
Vector.sort objects/iteration:6: 0.00ms
Vector.sort objects/iteration:7: 0.00ms
Vector.sort objects/iteration:8: 0.00ms
Vector.sort objects/iteration:9: 0.00ms
```

`Vector.sort` is _suspiciously_ fast. I'm not sure if it's not inlining somehow magically. I'm not sure if we want to spend too much time investigating this right now (although a nice comparison of performance of these two would be valuable).
I think it may still be worth to leave these benchmarks as-is - they can serve as a good starting point to investigate.